### DISCUSSION TITLE
I Created A Tool That Enables Users To Create Custom Actions Using Python! ✨

### DISCUSSION THREAD

**Discussion by lnxpy** [community/community#134129] [Actions]
### Select Topic Area

Show & Tell

### Hi community! 👋🏻

I’m excited to introduce an open-source project I’ve been working on in the last couple of months. This tool helps developers to create custom, reusable GitHub Actions using Python. There are some features that make it very handy! 🔥

🔗 Repository: [PyAction on GitHub](https://github.com/lnxpy/pyaction)
📄 Documentation: [PyAction Docs](https://pyaction.imsadra.me/)
📝 Walk-through Article: https://blog.imsadra.me/pyaction-create-github-actions-using-python

### Key Features 🔥 
- Run Actions Locally: Test your actions on your local machine before deploying them.
- Custom Bash Commands: Execute custom bash commands within the runner instance.
- Issue Forms Integration: Seamlessly read issues created with GitHub Issue Forms. (Very cool feature 🔥)

💯 Also, actions created with PyAction can be pushed to the GitHub Marketplace!

### Running Locally
![image](https://github.com/user-attachments/assets/ad81affd-f0e2-41ca-8a78-4158bc422f0f)

### Issue Forms Feature Demo ✨
In fact, with the Issue Forms feature, you can use GitHub's issue tab as the UI side for your Python codes. Consider the issue forms as regular UI forms that take a bunch of data through fields, process them, and return the data! [I have talked about this feature in detail here](https://pyaction.imsadra.me/tutorial/#issueform).

This is just an example of what you can create with the Issue Forms feature. You fill the fields and it runs a Python code that reads the information you just entered and passes them to ChatGPT.

https://github.com/user-attachments/assets/1c18ff77-5a2c-4cb5-8995-63708ae33221

Check out the repository and documentation to get started. I’m looking forward to your feedback and contributions. ❤️

**Comment by mdiktushar** [comment]
great work my friend. Keep going.

**Reply by lnxpy** [reply]
Thanks, man!

**Comment by lnxpy** [comment]
I would love for it to be seen and tried out by others. This forum is the best place I’ve found to discuss the tool, and I’m tagging you experts to explore its potential for further development. Your feedback would be incredibly valuable to me on this journey! 🙂

Feel free to let me know if you need any further adjustments or additions! So far, if you've found this mention bothering you, I am so sorry. I didn't mean it. 💔

CC: @davevad93 @queenofcorgis @ghostinhershell @ebndev

**Reply by davevad93** [reply]
Hey @lnxpy no bother whatsoever, I've read the documentation and it looks interesting, I'll give it a try! 👍

**Reply by lnxpy** [reply]
Thank you, [Davide](https://github.com/davevad93)!
I'm so glad you found it useful. :)

**Reply by grandmas-favorite** [reply]
Very cool @lnxpy! 😄

**Reply by lnxpy** [reply]
Thank you, @grandmas-favorite. :))
